compute_endpoint/tests/unit/test_endpointmanager_unit.py

```diff
@@ -789,3 +821,27 @@ def test_all_files_closed(successful_exec):
     assert mock_os.dup2.call_count == 3, "Expect to close 3 std* files"
     closed = [std_to_close for (_fd, std_to_close), _ in mock_os.dup2.call_args_list]
     assert closed == [0, 1, 2]
+
+
+@pytest.mark.parametrize(
+    "conf_size", [(True, 10), (True, 222), (False, 223), (False, 300)]
+)
+def test_pipe_size_limit(mocker, successful_exec, conf_size):
+    is_valid, conf_size = conf_size
+    _mock_os, _conf_dir, _mock_conf, _mock_client, em = successful_exec
+
+    conf_str = "$" * conf_size
+    # Add 34 bytes for dict keys, etc.
+    stdin_data_size = conf_size + 34
+
+    mock_log = mocker.patch(f"{_MOCK_BASE}log")
+    mocker.patch.object(fcntl, "fcntl", return_value=512)
+    mocker.patch(f"{_MOCK_BASE}render_config_user_template", return_value=conf_str)
+
+    with pytest.raises(SystemExit) as pyexc:
+        em._event_loop()
+
+    if is_valid:
+        assert pyexc.value.code == 86, "Q&D: verify we exec'ed, based on '+= 1'"
+    else:
+        assert f"{stdin_data_size} bytes" in mock_log.error.call_args[0][0]
```

Comment:
Took a question to understand why these don't _all_ pass: because 512 is what `fcntl.fcntl` returns (per the mock), but the code subtracts a constant 256 from that for general overhead &rarr; suggest a comment here, because that's an easy detail to miss.

Meanwhile, since 512 is hardcoded inside the test, `is_valid doesn't need to be specified but could be calculated.  Consider:
```python
def test_pipe_size_limit(mocker, successful_exec, conf_size):
    pipe_buffer_size = 512
    is_valid = pipe_buffer_size - conf_size - 256  # 256 == constant value removed by method
   ...
   mocker.patch.object(fcntl, "fcntl", return_value=pipe_buffer_size)
   ...
```

Reply:
All good points. I've incorporated these suggestions in a fresh commit.